### PR TITLE
Adds Trust Gambit Conditions for Tank in Party

### PR DESCRIPTION
**Pull request by neuromancerxi**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Adds Trust Gambit Conditions `PT_HAS_TANK` and `NOT_PT_HAS_TANK`
Adds Conditional Gambits for Volker to either DD or Tank depending on results

## Steps to test these changes

Log in a GM Test Char
`!changejob WAR 99`
`!addspell 903` -- Volker
`!addspell 910` -- Valaineral
Summon only Volker, Engage a mob
Volker will use Provoke, Defender, and Retaliation
Despawn or Defeat Mob
Summon Valaineral alongside of Volker, Engage a mob
Volker will use Berserk and Aggressor. Will use Provoke if Tank is < 50% Health

### CODE REVIEW

File: src/map/ai/helpers/gambits_container.cpp [hunk 1]
```diff
@@ -656,6 +656,16 @@ namespace gambits
...
                 break;
             }
+            case G_CONDITION::PT_HAS_TANK:
+            {
+                return PartyHasTank();
+                break;
+            }
+            case G_CONDITION::NOT_PT_HAS_TANK:
+            {
+                return (!PartyHasTank());
```
Comment (zach2good) [inline src/map/ai/helpers/gambits_container.cpp]: This does need parens around it

File: src/map/ai/helpers/gambits_container.cpp [hunk 2]
```diff
@@ -912,4 +922,18 @@ namespace gambits
         });
         return hasHealer;
     }
+    // used to check for tanks in party (Volker, AA Hume)
+    bool CGambitsContainer::PartyHasTank()
+    {
+        bool hasTank = false;
+        static_cast<CCharEntity*>(POwner->PMaster)->ForPartyWithTrusts([&](CBattleEntity* PMember) {
```
Comment (zach2good) [inline src/map/ai/helpers/gambits_container.cpp]: Formatting lamdas is hard and broken:

Drop the curly brace onto the next line, surround the whole statement with // clang-format off/on

**Review by zach2good — COMMENTED** [review]
Looks good apart from a little styling 👍

File: src/map/ai/helpers/gambits_container.cpp (hunk 1 above)
Comment (neuromancerxi) [reply src/map/ai/helpers/gambits_container.cpp]: Thanks for the review!

I'm a bit confused, would you mind clarifying this statement? Is the suggestion to wrap the return in parens (e.g.): 

`(return (!PartyHasTank()));`

or something else?

Comment (zach2good) [reply src/map/ai/helpers/gambits_container.cpp]: Just this:

`return !PartyHasTank();`

File: src/map/ai/helpers/gambits_container.cpp (hunk 2 above)
Comment (zach2good) [reply src/map/ai/helpers/gambits_container.cpp]: Here's an example:
```cpp

                // clang-format off
                PMaster->ForParty([](CBattleEntity* PMember)
                {
                    ((CCharEntity*)PMember)->PLatentEffectContainer->CheckLatentsPartyAvatar();
                });
                // clang-format on
```

**Review by zach2good — APPROVED** [review]

**Pull request merged**